www/class/centreonLDAP.class.php

```diff
@@ -585,8 +585,13 @@ public function search($filter, $basedn, $searchLimit, $searchTimeout)
         $sr = ldap_search($this->ds, $basedn, $filter, $attr, 0, $searchLimit, $searchTimeout);
 
         /* Sort */
-        $number_returned = ldap_count_entries($this->ds, $sr);
-        $this->debug("LDAP Search : " . (isset($number_returned) ? $number_returned : "0") . " entries found");
+        if ($sr !== false) {
+            $numberReturned = ldap_count_entries($this->ds, $sr);
+            $this->debug("LDAP Search : " . (isset($numberReturned) ? $numberReturned : "0") . " entries found");
+        } else {
+            $this->debug("LDAP Search : cannot retrieve entries");
```

Comment:
If you cannot retrieve entries, why not returning an empty array here ?

Reply:
updated